src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -97,7 +97,12 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
 
             if ($overwrite) {
                 if (!is_object($objectOrArray) && !is_array($objectOrArray)) {
-                    throw new UnexpectedTypeException($objectOrArray, 'object or array');
+                    throw new UnexpectedTypeException(sprintf(
+                        'Expected object or array, "%s" found when trying to write property "%s" of path "%s".',
+                        is_object($objectOrArray) ? get_class($objectOrArray) : gettype($objectOrArray),
```

Comment:
this condition makes no sense, since we already know that `$objectOrArray` is NOT an object